### ISSUE TITLE
Is it possible for your script to also export metadata?

### ISSUE THREAD

**Issue by sbholmes** [voruti/json2config#13]
There's also a `org.eclipse.smarthome.core.items.Metadata.json` file containing metadata about each item, if any exists.
Do you fancy also parsing that and add the relevant metadata onto the item it belongs to?

Here's an example snippet from that file on my server - the 'ga' is for google automation as per https://www.openhab.org/docs/ecosystem/google-assistant/

```
 "ga:NanoSwitch_HallLight_Switch": {
    "class": "org.eclipse.smarthome.core.items.Metadata",
    "value": {
      "key": {
        "segments": [
          "ga",
          "NanoSwitch_HallLight_Switch"
        ]
      },
      "value": "Light",
      "configuration": {}
    }
  }

```

So this would need to be added to the existing Item NanoSwitch_HallLight_Switch (as defined in the `org.eclipse.smarthome.core.items.Item.json` file) like this:

`NanoSwitch_HallLight_Switch { ga="Light" }` - pretty simple really.

**Comment by voruti** [comment]
No, currently, metadata is not taken into account.

For now, I won't introduce anything like that, but you - or anyone else - are of course welcome to create a pull request for it.

**Comment by sbholmes** [comment]
No worries, will send you a PR when i get a bit of free time - thanks!

**Comment by sbholmes** [comment]
Hello again
Coded something last night which I've just tested and seems to work, not quite ready for a PR yet.  I refactored ChannelAppender which does pretty much the same thing so I could re-use as much of the existing code as possible.  However you've got a couple of conditions that exclude groups - is that just for performance or does it cause issues when appending Channels?  Items and groups all have unique names - you can't have a group and item with the same name?  So I couldn't think of why you'd done it - none of the tests fail when I removed the two conditions:
https://github.com/voruti/json2config/blob/main/src/main/java/voruti/json2config/service/ChannelAppender.java#L89
https://github.com/voruti/json2config/blob/main/src/main/java/voruti/json2config/service/ChannelAppender.java#L114

Reason I ask is because you can add metadata to groups so I need to include groups in my MetadataAppender - while I can of course split things so Channel appending excludes groups but Metadata appending includes them, I just wanted to check that was necessary as it's easier to just modify the code so both appenders do not exclude groups.

**Comment by voruti** [comment]
Hm, I see. It was long ago; I don't remember exactly 😅.
I think I did this because groups never have channels, but - as you said - there is no difference for the generated output files, because groups never have channels 😂. You can just remove these exclusions, re-use the code and then both appenders do not exclude groups (just as you suggested).

**Comment by voruti** [comment]
Closed with #14.
